src/coreclr/minipal/Unix/doublemapping.cpp

```diff
@@ -74,6 +158,14 @@ bool VMToOSInterface::CreateDoubleMemoryMapper(void** pHandle, size_t *pMaxExecu
         return false;
     }
 
+#if defined(TARGET_AMD64) && !defined(TARGET_OSX)
+    if (!VerifyDoubleMapping(fd))
+    {
+        close(fd);
+        return false;
```

Comment:
Is this going to result in W^X being disable?

If I am reading the code correctly, it is going to result into startup failure.

Reply:
No, it is going to result in W^X being disabled - I've tested it in a x64 docker container on my M1. See the caller of the `VMToOSInterface::CreateDoubleMemoryMapper`:
https://github.com/dotnet/runtime/blob/745230568e52639e229dcc24c98fe0dbb4c1641d/src/coreclr/utilcode/executableallocator.cpp#L282-L298